src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.java

```diff
@@ -367,15 +367,10 @@ else if (state.currentScopeState == STATE_STATIC_VARIABLE_DEF) {
             }
         }
         else {
-            if (state.currentScopeState > STATE_STATIC_VARIABLE_DEF) {
-                if (!ignoreModifiers
-                        || state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF) {
-                    isStateValid = false;
-                    log(modifierAst, MSG_STATIC);
-                }
-            }
-            else {
-                state.currentScopeState = STATE_STATIC_VARIABLE_DEF;
+            if (state.currentScopeState > STATE_STATIC_VARIABLE_DEF && (!ignoreModifiers
```

Comment:
1. Please simplify this condition, it contains redundant check if you expand it
2. It becomes a single `if` under the last `else`, can we make it `else if (...) {}` instead of `else { if (...) {} }`?

Reply:
done

Reply:
1 - not done
if you expand that condition, you'll get this: `state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers || state.currentScopeState > STATE_STATIC_VARIABLE_DEF && state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF`, and it contains a redundant check, by removing it your initial condition becomes simpler.

Reply:
ok so we can simplify it as
```
        else if ((state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers)
                || state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF) {
            isStateValid = false;
            log(modifierAst, MSG_STATIC);
        }
```
I have pushed the changes.

Reply:
Now, we can put `state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF` first in the `if` as it is simper than than `state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers`. If a simple condition is true we don't need to evaluate a more complex condition.

Reply:
@rdiachenko simplifed
```
        else {
            if (state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF) {
                isStateValid = false;
                log(modifierAst, MSG_STATIC);
            }
            else if (state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers) {
                isStateValid = false;
                log(modifierAst, MSG_STATIC);
            }
        }
```
Although functionally it would not make any changes but yes readability will is far better than https://github.com/checkstyle/checkstyle/pull/13123#discussion_r1215668957.
 I also like this new implementation. should I update this?

Reply:
You can do just this:
```
else if (state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF
            || state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers) {
        isStateValid = false;
        log(modifierAst, MSG_STATIC);
}
```

Reply:
done